pyproject.toml

```diff
@@ -1,3 +1,130 @@
 [build-system]
-requires = ["setuptools", "wheel"]
-build_backend = "setuptools.build_meta"
+requires = ["poetry-core>=1.1.0"]
+build-backend = "poetry.core.masonry.api"
+
+
+[tool.poetry]
+name = "aea"
+version = "1.2.1"
+description = "Autonomous Economic Agent framework"
+authors = ["Fetch.AI Limited"]
+readme = "README.md"
+license = "Apache-2.0"
+homepage = "https://github.com/fetchai/agents-aea"
+classifiers = [
+    "Environment :: Console",
+    "Environment :: Web Environment",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft",
+    "Operating System :: Unix",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Communications",
+    "Topic :: Internet",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Software Development",
+    "Topic :: System",
+]
+
+[tool.poetry.scripts]
+aea = 'aea.cli:cli'
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.11"
+base58 = ">=1.0.3,<3.0.0"
+jsonschema = ">=3.0.0,<4.0.0"
+packaging = ">=20.3,<21.0"
+semver = ">=2.9.1,<3.0.0"
+protobuf = ">=3.19.0,<=3.19.4"
+pymultihash = "==0.8.2"
+pyyaml = ">=4.2b1,<6.0"
+requests = ">=2.22.0,<3.0.0"
+python-dotenv = ">=0.14.0,<0.18.0"
+ecdsa = ">=0.15,<0.17.0"
+
+pywin32 = { version = "==303", markers = "sys_platform == 'win32' or platform_system == 'Windows'" }
+
+# cli, test_tools extras
+click = { version = ">=7.0.0,<8.0.0", optional = true }
+
+
+[tool.poetry.extras]
+cli = ["click"]
+test_tools = ["click"]
+all=["click"]
```

Comment:
Do we need test_tools and all extras?

Reply:
Try to remove cli and test_tools if they are not explicitly used. They are redundant to all. 

We'll have:

- `pip install aea`: to get aea + all basic deps minus click
- `pip install aea[all]`: to get aea + all basic deps + click

Reply:
cli will stay till we update `docs/cli-how-to.md`

test_tools - removed